src/Bicep.Core/SemanticModel/Namespaces/AzNamespaceSymbol.cs

```diff
@@ -39,7 +39,15 @@ public class AzNamespaceSymbol : NamespaceSymbol
 
             // the use of FunctionPlacementConstraints.Resources prevents use of these functions anywhere where they can't be directly inlined into a resource body
             new FunctionOverload("reference", LanguageConstants.Object, 1, 3, Enumerable.Repeat(LanguageConstants.String, 3), null, FunctionFlags.RequiresInlining),
-            new FunctionWildcardOverload("list*", LanguageConstants.Any, 2, 3, new[] { LanguageConstants.String, LanguageConstants.String, LanguageConstants.Object }, null, new Regex("^list[a-zA-Z]+"), FunctionFlags.RequiresInlining),
+            new FunctionWildcardOverload(
+                "list*",
+                LanguageConstants.Any,
+                2, 3,
+                new[] { LanguageConstants.String, LanguageConstants.String, LanguageConstants.Object },
+                null,
+                new Regex("^list[a-zA-Z]+"),
+                new[] { "listKeys", "listApiKeys", "listAuthKeys", "listAdminKeys", "listSecrets", "listCredentials", "listSasTokens" },
```

Comment:
The list of common matching names is a lie for practically all resource types. With some additional types work, we should be able to pull in the real names that start with list* that are specific to that resource type.

Until the above is done, I think we should either not help with spelling of list* or produce an error message that says these aren't real functions but typical names.

Reply:
That's a good point. I'm more inclined toward to not show spelling suggestions for list* function. Showing error message saying these are typical names sounds a sub-optimal workaround to me. I'll remove the common matching names.

Reply:
The common matching names are removed.